DailyReport/src/main/java/app/controller/ReportController.java

```diff
@@ -0,0 +1,37 @@
+package app.controller;
+
+import java.util.Locale;
+
+import org.apache.log4j.Logger;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.MessageSource;
+import org.springframework.security.authentication.AnonymousAuthenticationToken;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.servlet.ModelAndView;
+
+import app.service.ReportService;
+
+@Controller
+public class ReportController extends BaseController{
+	private static final Logger logger = Logger.getLogger(HomeController.class);
+	
+	@RequestMapping(value = "/reports")
+	public ModelAndView showListReports() {
+		logger.info("Report List Page"); 
+		ModelAndView model = new ModelAndView("views/reports/report-list");
+		Authentication auth = SecurityContextHolder.getContext().getAuthentication();
+		String username = ""; 
+		if (checkAuth(auth)) {
+			username =  auth.getName();
+			model.addObject("currentUser", username);
+			model.addObject("noReportAlert", messageSource.getMessage("report.noreport", null, Locale.US)); 
+		}
+		else return new ModelAndView("redirect:/login");
```

Comment:
```suggestion
		if (!checkAuth(auth)) return new ModelAndView("redirect:/login");
		
		username =  auth.getName();
		model.addObject("currentUser", username);
		model.addObject("noReportAlert", messageSource.getMessage("report.noreport", null, Locale.US)); 
		
```
ri sẽ dễ đọc code hơn
tiêu chí là: cố gắng trả về (return) kết quả sớm nhất nếu có thể

Reply:
Em đã fix lại ở branch show-report rồi ạ